tests/templates/test_embeddings/test_amplitude.py

```diff
@@ -487,3 +544,32 @@ def test_torch(self, tol, features):
         res2 = circuit2(features)
 
         assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("features", all_features)
+    def test_autograd_pad_with(self, tol, features):
+        """Tests torch tensors and pad_with."""
+        import torch
+
+        features = torch.tensor(features, requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev, interface="torch")
+        def circuit_decomposed(features):
+            # need to cast to complex tensor, which is implicitly done in the template
+            state = qml.math.cast(
+                qml.math.hstack([features, qml.math.zeros_like(features)]), torch.complex128
+            )
+            qml.QubitStateVector(state, wires=range(4))
+            return qml.state()
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(x=None):
+            qml.AmplitudeEmbedding(x, list(range(4)), pad_with=0.0)
+            return qml.state()
+
+        res = circuit(features)
+        res2 = circuit_decomposed(features)
+
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
```

Comment:
It would be great to add a Jax test / Jax jit test

Reply:
The JAX-JIT already exists. It was added here: https://github.com/PennyLaneAI/pennylane/pull/3166